src/ParallelAlgorithms/LinearSolver/ConjugateGradient/ElementActions.hpp

```diff
@@ -31,11 +32,43 @@ class TaggedTuple;
 namespace LinearSolver::cg::detail {
 template <typename Metavariables, typename FieldsTag, typename OptionsGroup>
 struct ResidualMonitor;
+template <typename FieldsTag, typename OptionsGroup>
+struct UpdateOperand;
 }  // namespace LinearSolver::cg::detail
 /// \endcond
 
 namespace LinearSolver::cg::detail {
 
+// Helper class that actions below derive from to inherit the `is_ready`
+// function
+template <typename InboxTag, typename TemporalIdTag>
+struct Receive {
+  using inbox_tags = tmpl::list<InboxTag>;
+
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex>
+  static bool is_ready(const db::DataBox<DbTags>& box,
+                       const tuples::TaggedTuple<InboxTags...>& inboxes,
+                       const Parallel::GlobalCache<Metavariables>& /*cache*/,
+                       const ArrayIndex& /*array_index*/) noexcept {
+    const auto& inbox = get<InboxTag>(inboxes);
+    return inbox.find(db::get<TemporalIdTag>(box)) != inbox.end();
```

Comment:
Please just let me know how you would prefer this pattern be handled. The options I can think of are (in order of decreasing code duplication):

1. Copy-paste the `inbox_tags`, `is_ready` function and `extract_from_inbox` function to the 6 places in this PR where they are used.
2. Factor them into `detail` namespaces as is currently done in this PR.
3. Factor them into files in `Parallel/`, as is done in #2489.